internal/updater/providers/slickvpn/location.go

```diff
@@ -0,0 +1,117 @@
+package slickvpn
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/qdm12/gluetun/internal/models"
+	htmlutils "github.com/qdm12/gluetun/internal/updater/utils/html"
+	"golang.org/x/net/html"
+)
+
+const (
+	ColContinent = 1
+	ColCountry   = 2
+	ColCity      = 3
+	ColConfig    = 4
+)
+
+var (
+	ErrExtractServerFromLink   = errors.New("unable to extract server string from link")
+	ErrNoMatchingServer        = errors.New("couldn't find matching server")
+	ErrOpenvpnLinkNodeNotFound = errors.New("couldn't find OpenVPN link node")
+)
+
+func fetchLocationInfo(
+	ctx context.Context,
+	client *http.Client,
+	indexURL string,
+) (serversOut []models.Server, err error) {
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, indexURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create HTTP request: %w", err)
+	}
+
+	response, err := client.Do(request)
+	if err != nil {
+		return nil, fmt.Errorf("HTTP request failed: %w", err)
+	}
+	defer response.Body.Close()
+
+	doc, err := html.Parse(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("an error occurred parsing response body: %w", err)
+	}
+
+	rows := htmlutils.GetNodesBySelector(doc, "table#location-table > tbody > tr")
+
+	rowCount := 0
+	for _, rowNode := range rows {
```

Comment:
nit you can pre-allocate the length of the servers since each of the row corresponds to one server. Also use the row index `i` (see below why)

```suggestion
	servers = make([]models.Server, len(rows))
	for i, rowNode := range rows {
```

And then instead of using:

```go
serversOut = append(serversOut, server)
```

use

```go
serversOut[i] = server
```

you could even directly assign to `serversOut[i]` saving the `server := models.Server{}` declaration.